querybook/tests/test_lib/test_elasticsearch/test_elasticsearch.py

```diff
@@ -0,0 +1,478 @@
+            get_data_doc_editors_by_doc_id_patch.start()
+        )
+        self.addCleanup(get_data_doc_editors_by_doc_id_patch.stop)
+
+    def _create_private_shared_datadoc_mock(self):
+        mock_doc = MagicMock(
+            id=self.DATADOC_ID,
+            environment_id=self.ENVIRONMENT_ID,
+            owner_uid=self.AUTHOR_UID,
+            public=False,
+        )
+        self.get_data_doc_editors_by_doc_id_mock.return_value = [
+            MagicMock(uid="alice"),
+            MagicMock(uid="charlie"),
+        ]
+        return mock_doc
+
+    def _create_private_datadoc_mock(self):
+        mock_doc = MagicMock(
+            id=self.DATADOC_ID,
+            environment_id=self.ENVIRONMENT_ID,
+            owner_uid=self.AUTHOR_UID,
+            public=False,
+        )
+        self.get_data_doc_editors_by_doc_id_mock.return_value = []
+        return mock_doc
+
+    def _create_public_datadoc_mock(self):
+        mock_doc = MagicMock(
+            id=self.DATADOC_ID,
+            environment_id=self.ENVIRONMENT_ID,
+            owner_uid=self.AUTHOR_UID,
+            public=True,
+        )
+        return mock_doc
+
+    def _get_mock_query_cell(self):
+        return MagicMock(
+            id=self.QUERY_CELL_ID,
+            context=self.QUERY_TEXT,
+            created_at=CREATED_AT_DT,
+            doc=self._create_private_shared_datadoc_mock(),
+            meta={
+                "engine": self.ENGINE_ID,
+                "title": self.QUERY_CELL_TITLE,
+            },
+        )
+
+    def _assert_equals_query_item(self, result, expected_result):
+        for key, expected_value in expected_result.items():
+            if key != "full_table_name":
+                self.assertEqual(result[key], expected_value)
+            else:  # full_table_name array order can change
+                self.assertEqual(set(result[key]), set(expected_value))
+
+        self.assertEqual(len(result.items()), len(expected_result.items()))
+
+    def setUp(self):
+        self._patch_get_query_engine_by_id()
+        self._patch_get_datadoc_editors_by_doc_id()
+
+
+class QueryCellTestCase(BaseQueryTestCase):
+    def setUp(self):
+        super(QueryCellTestCase, self).setUp()
+        self.mock_cell = self._get_mock_query_cell()
+
+    def test_query_cell_to_es(self):
+        result = query_cell_to_es(self.mock_cell, session=MagicMock())
+        expected_result = {
+            "id": self.QUERY_CELL_ID,
+            "query_type": "query_cell",
+            "title": self.QUERY_CELL_TITLE,
+            "data_doc_id": self.DATADOC_ID,
+            "environment_id": self.ENVIRONMENT_ID,
+            "author_uid": self.AUTHOR_UID,
+            "engine_id": self.ENGINE_ID,
+            "statement_type": ["SELECT"],
+            "created_at": CREATED_AT_EPOCH,
+            "full_table_name": ["main.test_table", "main.test_table_2"],
+            "query_text": self.QUERY_TEXT,
+            "public": False,
```

Comment:
there is a lot of repetition, can we have some kind of shared dictionary for the common fields?